### ISSUE TITLE
Redash query editor is buggy on iPad

### ISSUE THREAD

**Issue by susodapop** [getredash/redash#4585]
### Issue Summary
I used Redash for a couple weeks from my iPad Pro (currently the latest model sold by Apple). I switched between finger inputs, external keyboards, and the Pencil. Here are the bugs I found in the query editor:

## There is no way to select text

If you double-tap a word it will be selected. But it does not show the characteristic "handles" that allow for expanding the text selection.

iOS 12 introduced a trick where holding the space bar invokes a "trackpad" like insertion point control. Invoking this feature within the query editor causes the insertion point to spring to the top of the text box. No text is selected.

You can work around this with an external keyboard using option, command, and the arrow keys. Non-keyboard users are out of luck.

## You can only backspace one word at a time

Related to the issue above. Hold-to-repeat is disabled for the backspace key. The fastest way to delete a portion of a query is to double-tap a word and then backspace it recursively. This is not feasible for anything but the smallest queries.

## The text insertion context menu is disabled

Related to the issues above. Since you can't select text there is no way to invoke the context menu with Cut/Copy/Paste/Undo/Redo shortcuts. This means you can't even write your query in another text-editor and paste it to Redash unless you know the three-finger paste gesture (see below).

## Three-finger paste gesture is inconsistent

Normally this feature is rock solid on iOS. But it misbehaves so often in Redash that it's effectively useless. Three-finger swipes are almost always interpreted as scroll gestures (see below).

## Three-finger undo gesture doesn't work

Related to above and below. The only undo that works is to physically shake the device.

## Most touches are treated as scroll inputs

Generally this is how iOS ought to be behave. But the Redash editor is especially picky about what it calls a "tap" and what it treats as a "scroll". Unless your hand is perfectly steady when tapping a word or a blankspace, the insertion point  move. Even with the Pencil and a steady hand it can take 5-10 attempts to move the insertion point from one side of a letter to the other. This is not an issue in any other text editing window I've seen on iOS.

## Can't resize the query editor

Related to the above. The drag-handles don't work on iOS.

## Query editor doesn't scroll to show insertion point

The onscreen keyboard on iPad consumes half the screen. If you invoke the onscreen keyboard and start typing, you need to manually scroll the query editor text box to see what you have entered. Most other apps will automatically scroll to do this.

## Can't scroll the view port

Related to the above. Most touch inputs are treated as scroll behavior. Which makes it near impossible to scroll down to see a visualization or the query refresh schedule.

### Technical details:

* Redash Version: Saas (V9 beta)
* Browser/OS: Safari / Chrome / Firefox / Edge on iOS (all of them use the same renderer)
* How did you install Redash: Saas

**Comment by kravets-levko** [comment]
Looks like everything except of 

> Can't resize the query editor

is related to Ace editor. Can you please try to reproduce all that issues in a demo editor [here](https://ace.c9.io/) and tell us what doesn't work there as well?

**Comment by arikfr** [comment]
@kravets-levko also the "Can't scroll the view port" is an issue on our end. It's the wrong description (there is no scroll on this page), but the real problem is that the bottom of the view gets cut:

![image](https://user-images.githubusercontent.com/71468/72988924-ee267100-3df5-11ea-87fc-50c2368b015f.png)

Notice how the metadata about the query result is missing from view.

**Comment by kravets-levko** [comment]
So, I tried our Query Editor on my Android device and managed to reproduce almost all issues related to selecting text/copy/paste/etc. (except of gestures because my device doesn't use have them at all). Also, behavior of editor on Ace demo page is exactly the same. Also, I tried [Monaco](https://microsoft.github.io/monaco-editor/) editor - and it has the same issues on mobiles :cry:

**Comment by arikfr** [comment]
@kravets-levko does the cut off reproduce too?

**Comment by kravets-levko** [comment]
No, but it's okay - my smartphone is too small so there are no sections but a regular page scroll (our layout for small screens)

**Comment by arikfr** [comment]
Got it. I think it's something specific to Safari, so not sure it will reproduce on another device - even with same screen size. Will try to investigate some more.

**Comment by naylanovo** [comment]
Hello there, I'm trying to login onto my Redash account in my iPad and it is absolutely impossible! the message is "wrong email or password" 

I tried it on Safari and Google Chrome app. Can you please help me? 

Thank you!

**Comment by susodapop** [comment]
@naylanovo that doesn't sound related to this issue. Please post your question on the user forum instead: discuss.redash.io

**Comment by susodapop** [comment]
Hi @naylanovo your issue looks unrelated. Login to Redash from iOS (both iphone and ipad) is currently supported and functional.
